src/views/token-swap/components/swap-form/swap-form.view.jsx

```diff
@@ -36,36 +58,97 @@ function SwapForm ({
   }, [accounts])
 
   React.useEffect(() => {
-    const from = accounts?.data.accounts.find(a => a.accountIndex === fromQuery)
-    const to = accounts?.data.accounts.find(a => a.accountIndex === toQuery)
-    setSelectedTokens({
-      [AmountBoxPosition.FROM]: from,
-      [AmountBoxPosition.TO]: to
-    })
+    const from = accounts.data?.accounts.find(a => a.accountIndex === fromQuery)
+    const to = accounts.data?.accounts.find(a => a.accountIndex === toQuery)
+    if (from && to) {
+      setSelectedTokens({
+        [AmountBoxPosition.FROM]: from,
+        [AmountBoxPosition.TO]: to
+      })
+    }
   }, [accounts])
 
+  React.useEffect(() => {
+    if (quotes.data?.quotes) {
+      const from = BigNumber.from(quotes.data.quotes[0].amountFromToken)
+      const to = BigNumber.from(quotes.data.quotes[0].amountToToken)
+      setAmountFrom(from)
+      setAmountTo(to)
+      if (positionUpdated === AmountBoxPosition.TO) {
+        setDefaultValues({
+          ...defaultValues,
+          [AmountBoxPosition.FROM]: { amount: from }
+        })
+      } else {
+        setDefaultValues({
+          ...defaultValues,
+          [AmountBoxPosition.TO]: { amount: to }
+        })
+      }
+    }
+  }, [quotes])
+
+  const onHandlerChange = (value, position) => {
```

Comment:
Well, technically the handler is not going to change hahaha I would call it something more `handleAmountChange` or similar.